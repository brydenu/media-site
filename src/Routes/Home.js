import React from "react";
import Card from "../Common/Card";
import Searchbar from "../Common/Searchbar";

export default function Home() {
    const headerContent = (
        <h2 className="home-title">
            Welcome, use the search bar to find media!
        </h2>
    );

    const bodyContent = <Searchbar />;
    return (
        <div className="main home">
            <Card header={headerContent} body={bodyContent} type="home" />
        </div>
    );
}
