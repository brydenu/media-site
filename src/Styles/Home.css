.main-home {
    width: 100%;
    display: flex;
    flex-flow: column nowrap;
    align-content: center;
    margin-top: 20px;
}

.card-header {
    display: flex;
    justify-content: center;
    margin: 10px;
}

.card-body {
    display: flex;
    justify-content: center;
}

.results-all {
    display: flex;
    flex-flow: column nowrap;
}

.movies-wrapper {
    display: flex;
    flex-flow: column nowrap;
    border: 2px solid blue;
}

.shows-wrapper {
    display: flex;
    flex-flow: column nowrap;;
    border: 2px solid green;
}

.songs-wrapper {
    display: flex;
    flex-flow: column nowrap;
    border: 2px solid purple;
}

.movies-results {
    display: flex;
    flex-flow: row wrap;
}

.shows-results {
    display: flex;
    flex-flow: row wrap;
}

.songs-results {
    display: flex;
    flex-flow: row wrap;    
}

.group-title {
    align-self: center;
}
