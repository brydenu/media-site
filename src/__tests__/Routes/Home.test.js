import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "../../Routes/Home";
import TestContext from "../../contextForTests";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <TestContext>
                <Home />
            </TestContext>
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <TestContext>
                <Home />
            </TestContext>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
