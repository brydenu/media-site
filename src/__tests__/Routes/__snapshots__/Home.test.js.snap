// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main main-home"
  >
    <div
      class="card-wrapper home"
    >
      <div
        class="card-content home"
      >
        <div
          class="card-header "
        >
          <h1
            class="home-title"
          >
            Welcome to the Media Site!
          </h1>
          <form
            class="search search-home"
          >
            <h2
              class="searched-header-title"
            >
              Use the search bar to find media
            </h2>
            <input
              autocomplete="off"
              name="q"
              placeholder=""
              type="text"
              value=""
            />
            <button
              class="searchbar-btn"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
        <div
          class="card-body home"
        >
          <div
            class="hidden-home-btns home-body"
          >
            <h3
              class="home-or"
            >
              or
            </h3>
            <div
              class="home-buttons"
            >
              <p>
                <a
                  class="home-link"
                  href="/login"
                >
                  Log in
                </a>
                or
                <a
                  class="home-link"
                  href="/signup"
                >
                  Sign up
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
